lib/client/index.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+var _ = require('lodash')
+  , q = require('q')
+  , defaultConfig = require('../config')
+  , Rx = require('rx')
+  , DataManager = require('./data-manager');
+
+
+var $fh, initialized = false, syncNotificationStream, syncNotifyListeners = [], mediator;
+
+/**
+ *
+ * Adding a single listener to be executed whenever a $fh.notify handler is executed.
+ *
+ * @param listener
+ */
+function addListener(listener) {
+  syncNotifyListeners.push(listener);
+}
+
+/**
+ * Initialising the $fh.sync client and registering a notification handler.
+ *
+ * The notification handler is an observable stream to allow different data sets to subscribe to notifications
+ * that are specific to them.
+ *
+ * This is because the $fh.sync client API has a single notification handler for all data sets.
+ *
+ * E.g. the workorders data set is only interested in notifications related to the workorders data set.
+ *
+ * (For more information on the $fh.sync Client API, see https://access.redhat.com/documentation/en/red-hat-mobile-application-platform-hosted/3/paged/client-api/chapter-11-fhsync)
+ *
+ * @param _$fh - The feedhenry client API.
+ * @param _syncOptions - The options to configure the sync client. See link above for available options.
+ * @param _mediator - The mediator to use for publishing topics.
+ */
+function init(_$fh, _syncOptions, _mediator) {
+  if (initialized) {
+    console.log('sync-client already initalized.');
+    return;
+  }
+
+  //Assigning the mediator
+  mediator = _mediator;
```

Comment:
Yep, good point. I will add some more detail.

I  was thinking out loud in my code.